apply-rule/src/main/scala/org/xalgorithms/apply_rules/Job.scala

```diff
@@ -1,17 +1,105 @@
 package org.xalgorithms.apply_rules;
 
+import com.mongodb.spark.MongoSpark
+import com.mongodb.spark.config._
 import config.Settings
 import kafka.serializer.StringDecoder
 import utils.SparkUtils._
 import utils.KafkaSinkUtils._
 import org.apache.spark.streaming.kafka.KafkaUtils
 import org.apache.spark.streaming.{Seconds, StreamingContext}
-import com.datastax.spark.connector.streaming._
 import org.xalgorithms.apply_rules.udt._
-import org.xalgorithms.apply_rules.udt.MyJsonProtocol._
-import spray.json._
+import org.apache.commons.lang3.reflect.FieldUtils
+
 
 object Job {
+  val VALID_SECTIONS = Array(
+    "envelope",
+    "item"
+  )
+
+  def executeRule(doc: Invoice, rule: Rule) = {
+    val ruleItems = rule.items
+    ruleItems.foreach({r =>
+      val whens = r.whens
```

Comment:
What do you need from the whens statements while executing the rule? This will have been used to *filter* the rule earlier in the pipeline. It doesn't contain anything for the execution of the rule.

Reply:
This patch is redundant, I'm going to abandon it soon